pandas/core/dtypes/cast.py

```diff
@@ -439,6 +444,59 @@ def changeit():
     return result, False
 
 
+def maybe_casted_values(index, codes=None):
```

Comment:
wouldn't object to typing things, but can also be a fllowon

Reply:
Will do a typing follow-on PR (for this method and a couple of other methods in `cast.py`)

Reply:
xref #37024